addons/mail/models/discuss/mail_thread.py

```diff
@@ -0,0 +1,31 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from markupsafe import Markup
+
+from odoo import models, _
+
+
+class MailThread(models.AbstractModel):
+    _inherit = "mail.thread"
+
+    def _notify_thread(self, message, msg_vals=False, **kwargs):
+        rdata = super()._notify_thread(message, msg_vals, **kwargs)
+        self._notify_mentioned_channel(message, **kwargs)
+        return rdata
+
+    def _notify_mentioned_channel(self, message, channel_ids=None, **kwargs):
+        """ Notify channels that have been mentioned in message"""
+        if channel_ids and self._name != "discuss.channel":
```

Comment:
Would remove the check on _name and make a proper inheritance in discuss.channel model that just "return" to skip doing anything. mail_thread should be model agnostic as much as possible :)

Reply:
Hello @tde-banana-odoo 
I have changed code according to @alexkuhn 's <a href="https://github.com/odoo/odoo/pull/162401#discussion_r1761140571">request</a>

Reply:
That's the base of modular approach. Generic behavior, overridden by model-based code. Otherwise when reading discuss.channel code you don't have all its business and model features, just part of it, not knowing part of it is hidden in other code places.

Model code should always be defined on models, that way models are complete and understandable. Having to grep methods of mixins is common usage: that's the purpose of mixins, propose generic code, then models inheriting form it may change it. Otherwise don't use mixins :) .